pkg/agent/proxy/proxier.go

```diff
@@ -1244,3 +1257,109 @@ func NewDualStackProxier(
 
 	return &metaProxierWrapper{ipv4Proxier, ipv6Proxier, metaProxier}, nil
 }
+
+func getServiceListOptionsModifier(serviceProxyName string) (func(*metav1.ListOptions), error) {
+	var serviceProxyNameSelector, nonHeadlessServiceSelector *labels.Requirement
+	var err error
+	nonHeadlessServiceSelector, err = labels.NewRequirement(corev1.IsHeadlessService, selection.DoesNotExist, nil)
+	if err != nil {
+		return nil, err
+	}
+	if serviceProxyName == "" {
+		serviceProxyNameSelector, err = labels.NewRequirement(labelServiceProxyName, selection.DoesNotExist, nil)
+	} else {
+		serviceProxyNameSelector, err = labels.NewRequirement(labelServiceProxyName, selection.DoubleEquals, []string{serviceProxyName})
+	}
+	if err != nil {
+		return nil, err
+	}
+	labelSelector := labels.NewSelector()
+	labelSelector = labelSelector.Add(*serviceProxyNameSelector, *nonHeadlessServiceSelector)
+	return func(options *metav1.ListOptions) {
+		options.LabelSelector = labelSelector.String()
+	}, nil
+}
+
+// CreateProxier creates IPv4 and IPv6 proxier, and starts the informerFactory used in proxiers.
+func CreateProxier(hostname string,
+	k8sClient clientset.Interface,
+	ofClient openflow.Client,
+	routeClient route.Interface,
+	v4Enabled bool,
+	v6Enabled bool,
+	nodePortAddressesIPv4 []net.IP,
+	nodePortAddressesIPv6 []net.IP,
+	proxyConfig antreaconfig.AntreaProxyConfig,
+	v4GroupCounter types.GroupCounter,
+	v6GroupCounter types.GroupCounter,
+	nestedServiceSupport bool,
+	stopCh <-chan struct{}) (Proxier, error) {
+	proxyAllEnabled := proxyConfig.ProxyAll
+	skipServices := proxyConfig.SkipServices
+	proxyLoadBalancerIPs := *proxyConfig.ProxyLoadBalancerIPs
+	serviceProxyName := proxyConfig.ServiceProxyName
+	optsModifier, err := getServiceListOptionsModifier(serviceProxyName)
+	if err != nil {
+		return nil, err
+	}
+	proxyInformerFactory := informers.NewSharedInformerFactoryWithOptions(k8sClient, informerDefaultResync, informers.WithTweakListOptions(optsModifier))
```

Comment:
This incurs 3 extra stream connections between **_each_** antrea-agent and kube-apiserver, which is not worth. It should use the original informer factory and just ignore services it's not interested in event handlers.

Reply:
Removed the informerFactory and added a label selector in event handlers.